File: engine/source/editor/editor/editor.cpp
#include "editor.h"

#include <iostream>

namespace moyu
{
    
    bool EditorApp::Init()
    {
        std::printf("Hello Editor Init\n");

        return true;
    }

    void EditorApp::Exit() { std::printf("Hello Editor Exit\n"); }

    bool EditorApp::Load()
    {
        std::printf("Hello Editor Load\n");

        return false;
    }

    void EditorApp::Unload() { std::printf("Hello Editor Unload\n"); }

    void EditorApp::Update(float deltaTime)
    {
        std::cout << "Hello Editor Update DeltaTime:" << deltaTime << std::endl;
    }

    void EditorApp::Draw() { std::printf("Hello Editor Draw\n"); }

} // namespace moyu

File: Engine/Source/Editor/Editor.cs

using System;
using MoyuEngine;

namespace MoyuEditor
{
    internal class Editor : Application
    {
        static void Main(string[] args)
        {
            Editor editor = new Editor();
            editor.Run();
        }

        public override void Exit()
        {
            Console.WriteLine("Hello, Exit!");
        }

        public override bool Init()
        {
            Console.WriteLine("Hello, Init!");
            return true;
        }

        public override void Update(float deltaTime)
        {
            Console.WriteLine($"Hello, Update {deltaTime}!");
        }
    }
}
